src/map.cpp

```diff
@@ -3925,53 +4377,57 @@ void map::remove_trap( const tripoint &p )
     }
 }
 /*
- * Get wrapper for all fields at xy
+ * Get wrapper for all fields at xyz
  */
-const field &map::field_at( const int x, const int y ) const
+const field &map::field_at( const tripoint &p ) const
 {
-    if( !inbounds( x, y ) ) {
+    if( !inbounds( p ) ) {
         nulfield = field();
         return nulfield;
     }
 
     int lx, ly;
-    submap *const current_submap = get_submap_at( x, y, lx, ly );
+    submap *const current_submap = get_submap_at( p.x, p.y, p.z, lx, ly );
```

Comment:
Had a bit of "branching hell" there for a while. Some of the changes are from earlier branches (before I created some of the useful overloads).